.github/workflows/macos.yml

```diff
@@ -92,6 +134,10 @@ jobs:
         if: ${{ matrix.test_task == 'check' && matrix.skipped_tests != '' }}
         continue-on-error: ${{ matrix.continue-on-skipped_tests || false }}
 
+      - name: Launchable - record tests
+        run: launchable record tests --flavor os=${{ matrix.os }} --flavor test_task=${{ matrix.test_task }} raw ${{ steps.setup_launchable.outputs.random_number }}.json
+        if: ${{ always() && github.repository == 'ruby/ruby' && (matrix.test_task == 'check' || matrix.test_task == 'test-all') }}
```

Comment:
Could you explain what the `always()` mean here? I am not familiar with this syntax. And I found one usage of the `always()` with `needs` syntax. But in this line, I don't see the `needs` syntax.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds

Reply:
Perhaps, the `always()` means even when a step fails before this step "Launchable - record tests", this step is still always executed?

Reply:
OK. I confirmed the "Launchable - record tests" step was still executed when I added a step to return the exit status non-zero before the step "Setup Launchable".
https://github.com/junaruga/ruby/actions/runs/7921897906/job/21628236404?pr=2#step:15:14

```
launchable record tests --flavor os=macos-12 --flavor test_task=check raw .json
...
/Users/runner/work/_temp/9ce83593-d4fb-4525-97c5-62cd8409b202.sh: line 1: launchable: command not found
Error: Process completed with exit code 127.
```

Reply:
You're right. `always()` execute the step even when canceled.

https://docs.github.com/en/actions/learn-github-actions/expressions#always

Reply:
All right. Thank you for introducing the document!